### ISSUE TITLE
Transformer tutorial multiplying with sqrt(d_model)

### ISSUE THREAD

**Issue by RogerJL** [pytorch/tutorials#2849]
https://github.com/pytorch/tutorials/blob/5e772fa2bf406598103e61e628a0ca0b8e471bfa/beginner_source/translation_transformer.py#L135

src = self.embedding(src) * math.sqrt(self.d_model)

shouln't this be

src = self.embedding(src) / math.sqrt(self.d_model)

at least that is the impression I got when reading the "Attention is all you need" paper.
Or is there some new research finding that multiplying is better?

cc @sekyondaMeta @svekars @kit1980 @subramen @albanD

**Comment by kangkannnng** [comment]
+1, also confused with that.

**Comment by kit1980** [comment]
Looking online, there are some discussions and looks like `*` is actually correct.
https://stackoverflow.com/questions/56930821/why-does-embedding-vector-multiplied-by-a-constant-in-transformer-model

I'm closing this, if someone more knowledgeable wants to correct me - please re-open.

**Comment by RogerJL** [comment]
Noted that this scaling was in a different place than the scaling in "Attention is all you need"
When rereading the paper, I noticed this hidden in 3.4 Embeddings and Softmax text

"In the embedding layers, we multiply those weights by √dmodel."

Closing is correct!
